src/frequenz/sdk/timeseries/_ringbuffer/buffer.py

```diff
@@ -108,6 +108,17 @@ def gaps(self) -> List[Gap]:
         """
         return self._gaps
 
+    def has_value(self, sample: Sample[QuantityT]) -> bool:
+        """Check if a sample has a value and it's not NaN.
+
+        Args:
+            sample: sample to check.
+
+        Returns:
+            True if the sample has a value and it's not NaN.
+        """
+        return not ((sample.value is None) or sample.value.isnan())
```

Comment:
If you want to save some parenthesis (which might make it easier to read):

```suggestion
        return sample.value is not None and not sample.value.isnan()
```

Reply:
Was thinking about that and found it even less readable. But removed one level.